torch/nn/utils/_stateless.py

```diff
@@ -1,120 +1,18 @@
-import contextlib
-from typing import Any, Callable, Dict, Iterator, List, Tuple
+# This file is never automatically imported within PyTorch so it is ok to
+# always warn here
+import warnings
 
-import torch
-from torch import Tensor
+warnings.warn("The `torch.nn.utils._stateless` code is deprecated now that "
+              "it is publicly available. Please use `torch.nn.utils.stateless "
+              "instead.", DeprecationWarning)
```

Comment:
If one imports this file multiple times, then it only gets imported once, right? If that is the case then I feel like it would be more robust to wrap the functions in a warning.

e.g.
```
def functional_call(*args, **kwargs):
  warnings.warn(...)
  return torch.nn.utils.stateless.functional_call(*args, **kwargs)
```

The thing I'm concerned about is: how do we guarantee that PyTorch doesn't import torch.nn.utils._stateless by accident between now and the release? Code review?

Reply:
Oh, you have a test for this. Okay I think we're good then

Reply:
Actually I can use the same test to make sure that we don't raise this warning when importing `torch`, very good point!